cabal-install/src/Distribution/Client/CmdRepl.hs

```diff
@@ -195,19 +189,52 @@ replCommand = Client.installCommand {
 -- "Distribution.Client.ProjectOrchestration"
 --
 replAction :: NixStyleFlags (ReplOptions, EnvFlags) -> [String] -> GlobalFlags -> IO ()
-replAction flags@NixStyleFlags { extraFlags = (replFlags, envFlags), ..} targetStrings globalFlags = do
-    let
-      with                 = withProject    cliConfig                   verbosity targetStrings
-      without globalConfig = withoutProject (globalConfig <> cliConfig) verbosity targetStrings
-
-    (baseCtx, targetSelectors, finalizer, replType) <-
-      withProjectOrGlobalConfig verbosity ignoreProject globalConfigFlag with without
-
-    when (buildSettingOnlyDeps (buildSettings baseCtx)) $
+replAction flags@NixStyleFlags { extraFlags = (replFlags, envFlags), ..} targetStrings globalFlags
+  = withContextAndSelectors AcceptNoTargets "repl:" (Just LibKind) flags targetStrings globalFlags $ \targetCtx ctx targetSelectors -> do
+    when (buildSettingOnlyDeps (buildSettings ctx)) $
       die' verbosity $ "The repl command does not support '--only-dependencies'. "
           ++ "You may wish to use 'build --only-dependencies' and then "
           ++ "use 'repl'."
 
+    let projectRoot = distProjectRootDirectory $ distDirLayout ctx
+
+    (replType, baseCtx) <- case targetCtx of
+      ProjectContext     -> return (ProjectRepl, ctx)
+      GlobalContext      -> do
+        unless (null targetStrings) $
+          die' verbosity $ "'repl' takes no arguments or a script argument outside a project: " ++ unwords targetStrings
```

Comment:
So we cant do a `cabal repl scriptFileInsideAProjectDir.hs`? 😞 
It will be surprising to being able to do `cabal run` and `cabal build` over such script and no `cabal repl`